### PR TITLE
Allow common DDL commands in actions for H2

### PR DESCRIPTION
**Pull request by calherries**
Initially, when I enabled actions for H2 in https://github.com/metabase/metabase/pull/28583 I created a whitelist of commands that could be allowed. The whitelist was pretty short, and only included commands for editing rows of a table: UPDATE, DELETE, INSERT, MERGE and TRUNCATE. 

This PR adds more commands to the whitelist. DDL commands like CREATE and DROP might be useful to have in actions. This will also enable H2 to support CSV uploads, since we can reuse the `qp.writeback/execute-write-query!` function to create the tables for the CSV uploads.

Changes:
1. I've added a subset of possible DDL commands to the list that I'm sure won't expose us to security vulnerabilities and will cover almost all use cases. Of course we could always add more for advanced use cases.
2. I ran `GRANT ALTER ANY SCHEMA TO GUEST;` on `resources/sample-database.db.mv.db`. Without running this command, DDL commands run with the GUEST user (i.e. the user used for actions connections) would fail.
3. I added `GRANT ALTER ANY SCHEMA TO GUEST` to the SQL that creates test DBs, so any tests that need to run DDL commands in actions will be able to do so.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29826)

### CODE REVIEW

File: test/metabase/sample_data_test.clj [hunk 1]
```diff
@@ -147,3 +147,26 @@
...
+            (jdbc/execute! conn-spec "CREATE SCHEMA NEW_SCHEMA;")
+            (is (contains? (set (get-schemas)) "NEW_SCHEMA")))
+          (testing "drop schema"
+            (jdbc/execute! conn-spec "DROP SCHEMA NEW_SCHEMA;")
+            (is (not (contains? (get-schemas) "NEW_SCHEMA"))))
+          (testing "create table"
+            (is (not (contains? (get-tables) "NEW_TABLE")))
+            (jdbc/execute! conn-spec "CREATE TABLE NEW_TABLE (id INTEGER);")
+            (is (contains? (get-tables) "NEW_TABLE"))
+          (testing "drop table"
```
Comment (qnkhuat) [inline test/metabase/sample_data_test.clj:170]: tests for add column, drop column too?

**Review by qnkhuat — APPROVED** [review]
LGTM but I think you should add some tests for adding columns and deleting columns for completeness.

File: test/metabase/sample_data_test.clj (hunk 1 above)
Comment (calherries) [reply test/metabase/sample_data_test.clj:178]: Done [Add test for adding/removing columns](https://github.com/metabase/metabase/pull/29826/commits/eed8cad480534e612cf5c1f8ac685f43e258af81)

**Review by escherize — APPROVED** [review]
👍

**Pull request merged**